presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcModule.java

```diff
@@ -58,13 +58,15 @@ public void configure(Binder binder)
         binder.bind(JdbcConnector.class).in(Scopes.SINGLETON);
         configBinder(binder).bindConfig(JdbcMetadataConfig.class);
 
-        newExporter(binder).export(Key.get(JdbcClient.class, InternalBaseJdbc.class))
+        newExporter(binder).export(Key.get(JdbcClient.class, TheJdbcClient.class))
                 .as(generator -> generator.generatedNameOf(JdbcClient.class, catalogName));
+        newExporter(binder).export(Key.get(ConnectionFactory.class, TheConnectionFactory.class))
+                .as(generator -> generator.generatedNameOf(ConnectionFactory.class, catalogName));
     }
 
     @Provides
     @Singleton
-    @InternalBaseJdbc
+    @TheJdbcClient
```

Comment:
You could use `@TheJdbcClient @StatsCollecting` here, to ensure stats-collecting is what gets exported to JMS